read_write.go

```diff
@@ -156,53 +153,40 @@ func Encode(data interface{}, xmlAttributes []xml.Attr, ns string) ([]byte, erro
 	return xmlBytes, nil
 }
 
-// addNameSpace will check if a name space was passed and find the first tag
-// where it should be applied.
-// TODO: This should work on any depth and tag, including extensions. I guess
-// we must support multiple name spaces and aliases as well.
-func addNameSpace(document *xmltree.Element, alias string) {
-	if alias == "" {
-		return
+// addNameSpaceAlias will check each node/element in the XML tree and if the
+// node has an xml.Name.Space value set an alias will be created and then added
+// to all child nodes. The alias will only be setup for the root element.
+func addNameSpaceAlias(document *xmltree.Element, nsAdded bool) *xmltree.Element {
+	namespaceAliases := map[string]string{
```

Comment:
Not sure where this should be placed, guess it could be a list on the session and passed all the way via `Encode()`.